lib/fast_jsonapi/serialization_core.rb

```diff
@@ -41,13 +41,18 @@ def links_hash(record, params = {})
       end
 
       def attributes_hash(record, fieldset = nil, params = {})
-        attributes = attributes_to_serialize
-        attributes = attributes.slice(*fieldset) if fieldset.present?
+        attributes = slice_attributes(attributes_to_serialize, fieldset)
         attributes.each_with_object({}) do |(_k, attribute), hash|
           attribute.serialize(record, params, hash)
         end
       end
 
+      def slice_attributes(attributes, fieldset)
+        return attributes unless fieldset.present?
+        fieldset = fieldset.map { |key| run_key_transform(key) }
```

Comment:
This might not be performant to run on serialization of each object. I wonder if we can do all of this once per instantiation of the serializer.